fluent-resmgr/src/resource_manager.rs

```diff
@@ -92,24 +109,41 @@ impl ResourceManager {
         &self,
         locales: Vec<LanguageIdentifier>,
         resource_ids: Vec<String>,
-    ) -> impl Iterator<Item = FluentBundle<&FluentResource>> {
-        let res_mgr = self;
-        let mut idx = 0;
-
-        iter::from_fn(move || {
-            locales.get(idx).map(|locale| {
-                idx += 1;
-                let mut bundle = FluentBundle::new(vec![locale.clone()]);
-                for resource_id in &resource_ids {
-                    let resource = res_mgr.get_resource(resource_id, &locale.to_string());
-                    bundle.add_resource(resource).unwrap();
-                }
-                bundle
-            })
-        })
+    ) -> Result<impl Iterator<Item = FluentBundle<&FluentResource>>, Vec<ResourceManagerError>>
+    {
+        let mut ptr = 0;
+        let mut errors: Vec<ResourceManagerError> = vec![];
+        let mut res = vec![];
+
+        while let Some(locale) = locales.get(ptr) {
```

Comment:
Issue: This changes the behavior of this function from being lazy (where the bundles are loaded on each iteration),  to fully evaluating the bundles. The intent for this iterator is to only evaluate the bundle on iteration.

Rather than returning ` Result<Iterator>` this should return a:

`impl Iterator<Item = Result< ... >>`

Reply:
Wooops, sorry about that 🙈